includes/Services/SiteGenService.php

```diff
@@ -999,6 +999,61 @@ public static function get_nav_link_grammar_from_post_data( $id, $name, $url ) {
 		return "<!-- wp:navigation-link {\"label\":\"$name\",\"type\":\"page\",\"id\":$id,\"url\":\"$url\",\"kind\":\"post-type\"} /-->";
 	}
 
+	/**
+	 * Compresses an image based on its MIME type.
+	 *
+	 * This function takes raw image data and its MIME type as input, and
+	 * returns the compressed image data.
+	 *
+	 * @param string $image_data The raw data of the image to be compressed.
+	 * @param string $mime_type The MIME type of the image (e.g., 'image/jpeg', 'image/png').
+	 *
+	 * @return string|false Returns the compressed image data as a string if successful or false.
+	 */
+	public static function compress_image( $image_data, $mime_type ) {
+		/*
+		Steps : Create an image resource from string, check type, apply gd functions to reduce size based on quality,
+		save output buffer content and return it, clean output buffer and destry created iamge.
+		*/
+
+		$image = imagecreatefromstring( $image_data );
+		if ( false === $image ) {
+			error_log( 'Failed to create image resource.' );
```

Comment:
I think we are supposed to not use them anymore, Let's return errors, and as this compression is just an enhancement instead of resulting in errors let's continue execution as you've done.